src/audio/copier/copier.c

```diff
@@ -35,6 +35,7 @@
 #include <errno.h>
 #include <stddef.h>
```

Comment:
You are not just splitting host_prepare() here you are creating entire new files. I think we deserve a better commit message explaining things. 

First please describe what is the problem this commit is trying to solve.

Reply:
ok，this patch exactly do the same thing as you mentioned, only add host_prepare_dma, other changes are all for this adding.
Regarding the commit message, let me add more information.

Reply:
@btian1 you need to use the commit message and show the old API, why it's not suitable, the new API and why this supports the use case. Otherwise it's difficult to review as I dont understand why the change is needed or what it does.

Reply:
@lgirdwood @btian1 calling these functions a new API is a misnomer. Previously, a copier dev created a host component device to handle the DMA copies. But with this change, the copier dev handles DMA copies without the need for the host component device. So these new functions declared in host_copier.h are merely exposing the common functions that are used by both the copier and the host component

Reply:
ok, I will modify commit title for better understanding